src/query/sql/src/planner/plans/delete.rs

```diff
@@ -25,6 +26,8 @@ pub struct DeletePlan {
     pub database_name: String,
     pub table_name: String,
     pub selection: Option<ScalarExpr>,
+    // The case: selection is subquery
+    pub input_expr: Option<SExpr>,
```

Comment:
It's also needed by `delete` with using.

Such as `DELETE FROM films USING producers
  WHERE producer_id = producers.id AND producers.name = 'foo';`

`Using` will be transferred to cross join, and `input_expr` will be cross join